exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ProofMapIndexProxy.java

```diff
@@ -221,22 +221,21 @@ public UncheckedMapProof getProof(K key) {
    * Returns a proof that there are values mapped to the specified keys or that there are no such
    * mappings.
    *
-   * @param keyList list of proof map keys which might be mapped to some value, each must be 32-byte
-   *     long
+   * @param keys proof map keys which might be mapped to some values, each must be 32-byte long
    * @throws IllegalStateException if this map is not valid
    * @throws IllegalArgumentException if the size of any of the keys is not 32 bytes
    */
-  public UncheckedMapProof getProof(List<K> keyList) {
-    return nativeGetMultiProof(getNativeHandle(), mergeKeysIntoByteArray(keyList));
+  public UncheckedMapProof getProof(K... keys) {
```

Comment:
I prefer the first one

Reply:
Why? :upside_down_face:

Reply:
Because it's safe and doesn't bring a huge overhead :)
Also, how about:`getProof(key: K, otherKeys: K...)`

Reply:
Thanks! Forgot about the PM indeed.
I like this one
`
+ getProof(key: K)
+ getProof(key1 : K, key2 : K, otherKeys : K...)
`
and I'd also add `getProof(keys: List<K>)` right away

Reply:
> Also, how about:getProof(key: K, otherKeys: K...)

Yes, it would better than the first, thanks! We can internall check that `otherKeys` is empty. The only reason to use the original, with two overloads is efficiency: `getProof(key: K, otherKeys: K...)` always requires passing an array of varargs — but (1) the overhead is negligible _in our heavy-weight implementation of this method_; (2) one can imagine a runtime optimizing allocations of empty arrays — but I haven't verified that claim :-)

> and I'd also add getProof(keys: List<K>) right away

:+1:, `j.u.Collection<? extends K>`